src/main/java/org/chocosolver/solver/constraints/nary/cumulative/literature/ouelletQuimper2013/PropCumulative.java

```diff
@@ -0,0 +1,283 @@
+/*
+ * This file is part of choco-solver, http://choco-solver.org/
+ *
+ * Copyright (c) 2019, IMT Atlantique. All rights reserved.
+ *
+ * Licensed under the BSD 4-clause license.
+ *
+ * See LICENSE file in the project root for full license information.
+ */
+package org.chocosolver.solver.constraints.nary.cumulative.literature.ouelletQuimper2013;
+
+import org.chocosolver.solver.constraints.nary.cumulative.literature.CumulativeFilter;
+import gnu.trove.list.array.TIntArrayList;
+import org.chocosolver.solver.exception.ContradictionException;
+import org.chocosolver.solver.variables.IntVar;
+import org.chocosolver.solver.variables.Task;
+
+import java.util.*;
+
+/**
+ * Cumulative constraint filtering algorithms described in the following paper :
+ * Ouellet, P., Quimper, C.-G.: Time-table-extended-edge-finding for the cumulative constraint. In: Proceedings of the 19th International Conference on Principles and Practice of Constraint Programming (CP 2013), pp. 562-577 (2013). https://doi.org/10.1007/978-3-642-40627-0_42
+ *
+ * @author Arthur Godet <arth.godet@gmail.com>
+ * @since 23/05/2019
+ */
+public class PropCumulative extends CumulativeFilter {
```

Comment:
The class name should not be so generic, there are many other `PropCumulative` classes, that may be confusing.